pp-mock/handlers/transport/v3/transport_search.go

```diff
@@ -109,21 +109,21 @@ func (*TransportSearchV3Server) TransportSearch(ctx context.Context, req *transp
 				}, nil
 			}
 
-			if queryTrip.Departure == nil || queryTrip.Arrival == nil ||
-				queryTrip.Departure.Date == nil || queryTrip.Arrival.Date == nil ||
-				queryTrip.Departure.Location == nil || queryTrip.Arrival.Location == nil {
+			if queryTrip.Departure == nil ||
+				queryTrip.Departure.Date == nil ||
+				queryTrip.Departure.Location == nil {
 				return &transportv3.TransportSearchResponse{
 					Header: &typesv1.ResponseHeader{
 						Status: typesv1.StatusType_STATUS_TYPE_FAILURE,
 						Alerts: []*typesv1.Alert{{
-							Message: "Invalid trip filter: departure and arrival must be provided",
+							Message: "Invalid trip filter: departure must be provided",
 							Type:    typesv1.AlertType_ALERT_TYPE_ERROR,
 						}},
 					},
 				}, nil
 			}
 
-			if !queryTrip.Departure.Location.HasLocationCodes() || !queryTrip.Arrival.Location.HasLocationCodes() {
+			if !queryTrip.Departure.Location.HasLocationCodes() {
```

Comment:
This block now only checks for departure location code but the error message is 
> Unsupported trip filter: departure and arrival must provide location codes

Reply:
This has been reverted back and is part of the switch case error handling.